modules/fb_instant_articles_display/fb_instant_articles_display.module

```diff
@@ -0,0 +1,777 @@
+    }
+    foreach ($entity_types as $entity_type => $bundles) {
+      foreach ($bundles as $bundle_type => $bundle) {
+        $form_name = $bundle_type . '_node_form';
+        if ($form_id == $form_name) {
+          // Add the module's form settings to this individual node's form.
+          // Add a vertical tab to the node type form.
+          if (user_access('administer fb_instant_articles_display')) {
+            // Build fieldset for vertical tab section.
+            $fieldset = array(
+              '#type' => 'fieldset',
+              '#title' => t('Facebook Instant Articles'),
+              '#description' => t('Configure entity for Facebook Instant Article mappings.'),
+              '#group' => 'additional_settings',
+              '#collapsible' => TRUE,
+              '#collapsed' => TRUE,
+            );
+
+            // Has the section already been created (perhaps by a sub module)?
+            if (isset($form['fb_instant_articles_display'])) {
+              // Add additional fieldset data.
+              $form['fb_instant_articles_display'] += $fieldset;
+            }
+            // Else, create the fieldset.
+            else {
+              $form['fb_instant_articles_display'] = $fieldset;
+            }
+
+            // Is an Article?
+            $type = $form['#node']->type;
+            $fb_instant_enabled = fb_instant_articles_display_is_article($form['#node']->nid);
+
+            // Previously checked the product checkbox?
+            $previously_checked = (isset($form_state['values']) && $form_state['values']['fb_instant_articles_display']['fb_instant_enabled']);
+
+            // Build the product checkbox.
+            $form['fb_instant_articles_display']['fb_instant_enabled'] = array(
+              '#type' => 'checkbox',
+              '#title' => t('Include entity in Facebook Instant Articles feed.'),
+              '#description' => t('Enable this entity to be included in the Facebook Instant Articles feed.'),
+              '#weight' => 0,
+              '#default_value' => ($previously_checked || $fb_instant_enabled) ? TRUE : FALSE,
+            );
+
+            // Add custom submit.
+            $form['#submit'][] = 'fb_instant_articles_display_node_form_submit';
+          }
+          // Since only one node add/edit form can (probably?) be loaded on a
+          // page at a time, we can break out of this loop.
+          break;
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Submit callback for node form.
+ */
+function fb_instant_articles_display_node_form_submit($form, &$form_state) {
+  $fb_instant_enabled = fb_instant_articles_display_is_article_type('node', $form['type']['#value']);
+  if ($fb_instant_enabled) {
+    if (!isset($form_state['values']['fb_instant_enabled'])) {
+      $form_state['values']['fb_instant_enabled'] = 0;
+    }
+    $value = $form_state['values']['fb_instant_enabled'];
+    $default_enabled = variable_get('fb_instant_articles_display_all_enabled_default', TRUE);
```

Comment:
Right, also that function name has changed to fb_instant_articles_display_get_article_entity_types(), so we should either change that for now or remove that preconfigured filter altogether.